Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/core/impl/ConceptManager.java

```diff
@@ -417,20 +419,38 @@ public void storeModifiedConcept(ConceptEntry entry) throws LuceneException, Ill
 	}
 
     protected String generateId(String prefix) {
-        String id = prefix + UUID.randomUUID().toString();
-
         while (true) {
-            ConceptEntry example = null;
-            example = new ConceptEntry();
+            String id = prefix + generateUniqueId();
+
+            ConceptEntry example = new ConceptEntry();
             example.setId(id);
             // if there doesn't exist an object with this id return id
             List<Object> results = client.queryByExample(example);
             if (results == null || results.size() == 0)
                 return id;
+        }
+    }
 
-            // try other id
-            id = prefix + UUID.randomUUID().toString();
+    /**
+     * This methods generates a new 6 character long id. Note that this method
+     * does not assure that the id isn't in use yet.
+     * 
+     * Adapted from
+     * http://stackoverflow.com/questions/9543715/generating-human-readable
+     * -usable-short-but-unique-ids
+     * 
+     * @return 6 character id
+     */
+    private String generateUniqueId() {
+        char[] chars = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz".toCharArray();
+
+        Random random = new Random();
+        StringBuilder builder = new StringBuilder();
+        for (int i = 0; i < 6; i++) {
```

Comment:
@mkarthik90 Can you make the ids 12 characters long instead of 6. We need to make sure we have enough possible ids.